api/coordinatornetwork/coordinatornetwork.go

```diff
@@ -0,0 +1,118 @@
+/*
+Package coordinatornetwork implements a comunication layer among coordinators
+in order to share information such as transactions in the pool and create account authorizations.
+
+To do so the pubsub gossip protocol is used.
+This code is currently eavily based on this example: https://github.com/libp2p/go-libp2p/blob/master/examples/pubsub
+*/
+package coordinatornetwork
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/hermeznetwork/hermez-node/common"
+	"github.com/hermeznetwork/hermez-node/log"
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p/p2p/discovery"
+
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	mrand "math/rand"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+)
+
+const (
+	// port                = "3298"
+	discoveryInterval   = time.Hour // TODO: move to config file
+	discoveryServiceTag = "hermez-coordinator-network"
+)
+
+type CoordinatorNetwork struct {
+	txsPool  pubSubTxsPool
+	TxPoolCh chan *common.PoolL2Tx
+}
+
+// TODO: port should be constant, but this makes testing easier
+func NewCoordinatorNetwork(port string) (CoordinatorNetwork, error) {
+	ctx := context.Background()
+	// create a new libp2p Host that listens on a random TCP port
+	cfgOpts := libp2p.ListenAddrStrings("/ip4/0.0.0.0/tcp/" + port)
+	randNum, err := strconv.Atoi(port)
+	if err != nil {
+		return CoordinatorNetwork{}, err
+	}
+	// TODO: generate ID from coordinator's priv key
+	r := mrand.New(mrand.NewSource(int64(randNum)))
+	priv, _, err := crypto.GenerateKeyPairWithReader(crypto.RSA, 2048, r)
```

Comment:
Maybe here we can use the key already imported into the hermez node.

Reply:
Yeah it would be awesome to find a way to derivate the keys in a way that:

- Private key (BJJ or Ethereum) ==> libp2p private key (deterministic)
- Public key (BJJ or Ethereum) ==> libp2p public key (deterministic)
- Security is preserved

This way the network could allow to seamlessly communicate to specific coordinators through p2p by just reading on-chain data